### DISCUSSION TITLE
how to unclog the UI without grouping?

### DISCUSSION THREAD

**Discussion by alex1x** [concourse/concourse#7123] [Help & Support] [answered]
Hello,

I have a large pipeline that uses inputs and outputs shared between multiple jobs. These visually all go to the bottom of the pipeline and make everything hard to see (and render). Plus they don't offer anything useful (at least not in this case). The worst part is that the same thing (e.g. 'service-A-output') is repeated tens if not hundreds of times given a large pipeline where this output is consumed by many jobs. I would ideally have the option to just not show it at all in the UI, same for all these similar little blocks. This is how much duplication there is in the UI (that's just a tiny sample, the full extent is a lot worse and won't fit in a screen).

<img width="1385" alt="Screenshot 2021-06-03 at 12 19 33" src="https://user-images.githubusercontent.com/6977670/120636819-04baee80-c466-11eb-9100-52ac2a67d807.png">

Because all these little blocks are there, they take lots of screen real estate away from the pipeline and make the UI sluggish

<img width="370" alt="Screenshot 2021-06-03 at 12 27 58" src="https://user-images.githubusercontent.com/6977670/120637736-29639600-c467-11eb-88bb-a246af4241d1.png">

I realise the pipeline is large (about 30 thousand lines)... as a solution I use groups to look at smaller pieces of the pipeline, nevertheless this is something that could be improved in some way other than "just don't look at the whole pipeline view". 

Is there any general guidance on how to handle this? Alternatively is there any planned work on customizing the UI by e.g. setting a `visible` boolean flag or something on individual components (e.g. resources), or setting the whole pipeline to be "visually verbose" or not?

Thanks,
Alex

**Comment by taylorsilva** [comment, marked as answer]
Yeah the current pipeline view isn't very useful when your pipeline becames as large as yours. We don't have any plans to change this right now. As you already stated, creating groups is the best way to manage this right now.

One idea I had is that maybe there could be a slightly modified view where all the resources are collapsed into a single line between jobs. So instead of 4 resources going from jobA to jobB, you see one line going from jobA to jobB and clicking on the line would expand or take you to a new view displaying the resources between the two jobs.

**Reply by alex1x** [reply]
Thanks, yes your idea makes sense to me that would probably be a significant improvement already.
